kyo-stm/shared/src/main/scala/kyo/RefLog.scala

```diff
@@ -0,0 +1,25 @@
+package kyo
+
+import STM.internal.*
+
+opaque type RefLog = Map[TRef[Any], Entry[Any]]
+
+private[kyo] object RefLog:
+
+    given Tag[RefLog] = Tag[Map[TRef[Any], Entry[Any]]]
+
+    val empty: RefLog = Map.empty
+
+    extension (self: RefLog)
+
+        def put[A](ref: TRef[A], entry: Entry[A]): RefLog =
+            self.updated(ref.asInstanceOf[TRef[Any]], entry.asInstanceOf[Entry[Any]])
+
+        def get[A](ref: TRef[A]): Maybe[Entry[A]] =
+            val refAny = ref.asInstanceOf[TRef[Any]]
+            Maybe.when(self.contains(refAny))(self(refAny).asInstanceOf[Entry[A]])
```

Comment:
You can also do `Maybe(self.getOrElse(refAny, null))` to avoid the `contains` lookup

Reply:
I believe `getOrElse` will call `contains` anyway. I'd prefer to avoid `null`.